### DISCUSSION TITLE
Changing "Workflow permissions" to "Read and write"

### DISCUSSION THREAD

**Discussion by stefanpems** [community/community#57244] [Enterprise] [answered]
Hello.
In my personal account I could successfully change the "Workflow permissions" from "Read" to "Read and write" (under Repo -> Settings -> Actions/General).
I just created a trial Enterprise. There, at organization or repo level, I cannot make the same change: the option "Read and write" is disabled with the message "Repository administrators will only be able to change the default permissions to a more restrictive setting".
How can I ensure to assign also write permissions to the workflows in my organizational repos?
Thank you
 Stefano

**Comment by santoshD93** [comment]
I am facing the same issue.

**Comment by Ahhj93** [comment]
You have to change this in your organization.

**Reply by neuquen** [reply]
How?

**Reply by Ahhj93** [reply]
In the organisation page

Go to Settings → Actions → General 
At the bottom change to Read and write permissions and Save

**Reply by difhel** [reply]
this setting is locked
![image](https://github.com/community/community/assets/78644136/87700a9d-f8f5-48a2-98dd-44b8c48648e8)

**Reply by joshjohanning** [reply]
@difhel If this is a screenshot from the _organization page_ it looks like there's a policy in your enterprise to set it this way.  See https://docs.github.com/en/enterprise-cloud@latest/admin/policies/enforcing-policies-for-your-enterprise/enforcing-policies-for-github-actions-in-your-enterprise#enforcing-a-policy-for-workflow-permissions-in-your-enterprise

**Reply by techClick** [reply]
If this setting is locked in a repository. Its because it's locked in 'organization' level. If it's then locked in 'organization' level, its because it's locked in 'enterprise' level.

**Comment by 1aziz** [comment]
I'm having the same issue, and strangely enough, even on organisation level.

**Comment by jamieweavis** [comment]
We're running into the same issue, the `Read and write permissions` option in `Settings > Actions > General` is disabled for our organisation (and individual repos too)... We also just started an Enterprise trial, could be related?

**Reply by jamieweavis** [reply]
We figured it out, when upgrading to enterprise an additional layer of hierarchy is added to your GitHub, the setting is located in the top level which is now enterprises instead of organisations

- https://github.com/organizations/your-org-name-here/settings/actions 🚫
- https://github.com/enterprises/your-enterprise-name-here/settings/actions ✅

**Reply by joshjohanning** [reply]
Correct! It should be an [enterprise policy](https://docs.github.com/en/enterprise-cloud@latest/admin/policies/enforcing-policies-for-your-enterprise/about-enterprise-policies) now. Enterprise policies apply to all orgs in the enterprise.

**Comment by Solelands** [comment]
i get same issue as well ... is there any solution for this..?

**Comment by chrispat** [comment, marked as answer]
You shouldn't globally change this permission.  You should set the specific permissions for each workflow https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions.

**Reply by RKeertishKumar** [reply]
Thank you so much.

**Reply by theoturner** [reply]
This doesn't work for pull requests as of October 2025. The per-workflow permission in yaml worked prior to upgrading to GitHub Enterprise. After upgrading, you get error:
```
GraphQL: GitHub Actions is not permitted to create or approve pull requests (createPullRequest)
```
This occurs even with the per-workflow yaml configured. Enabling the enterprise-level setting fixes the issue:
https://github.com/enterprises/insert-your-enterprise-name-here/settings/actions

**Reply by JasonGross** [reply]
`workflows` is not listed as a valid permissions target for `permissions` on the [doc](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#defining-access-for-the-github_token-scopes):
<img width="747" height="516" alt="image" src="https://github.com/user-attachments/assets/8c3295b4-3c4b-4adf-a464-17259182f1dd" />

**Reply by joshjohanning** [reply]
@JasonGross right you'll have to use a GitHub app if you want to make modifications to workflows from another action, or trigger/chain workflows together. See : https://docs.github.com/en/actions/how-tos/write-workflows/choose-when-workflows-run/trigger-a-workflow#triggering-a-workflow-from-a-workflow

**Comment by PabloAndres33** [comment]
Thanks you very much, it helped me

**Comment by abdul-musabbir** [comment]
Number #1

**Comment by mdabdulmottalib** [comment]
as

[a comment by an automated account is not shown]
